File: bank-service/src/main/java/rs/edu/raf/dto/ExchangeInvoiceDTO.java
package rs.edu.raf.dto;

import java.math.BigDecimal;

public record ExchangeInvoiceDTO(Long id, BankAccountDTO senderAccount, BankAccountDTO receiverAccount, BigDecimal amount,
                                 CurrencyDTO senderCurrency, CurrencyDTO receiverCurrency, BigDecimal exchangeRate,
                                 BigDecimal profit, Long dateAndTime) {
}

File: bank-service-kotlin/src/main/kotlin/org/example/bankservicekotlin/dto/ExchangeInvoiceDTO.kt
package org.example.bankservicekotlin.dto

import java.math.BigDecimal

data class ExchangeInvoiceDTO(
    val id: Long?,
    val senderAccount: BankAccountDTO,
    val receiverAccount: BankAccountDTO,
    val amount: BigDecimal,
    val senderCurrency: CurrencyDTO,
    val receiverCurrency: CurrencyDTO,
    val exchangeRate: BigDecimal,
    val profit: BigDecimal,
    val dateAndTime: Long
)
